iothub_client/src/blob.c

```diff
@@ -13,32 +13,132 @@
 #include "azure_c_shared_utility/azure_base64.h"
 #include "azure_c_shared_utility/shared_util_options.h"
 
-#define HTTP_STATUS_CODE_OK                 200
-#define IS_HTTP_STATUS_CODE_SUCCESS(x)      ((x) >= 100 && (x) < 300)
-
 static const char blockListXmlBegin[]  = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\r\n<BlockList>";
 static const char blockListXmlEnd[] = "</BlockList>";
 static const char blockListUriMarker[] = "&comp=blocklist";
+static const char blockListLatestTagXmlBegin[] = "<Latest>";
```

Comment:
Instead of implementing blob upload ourselves, have we considered exposing the URI and SAS token to be used with one of the existing Azure Blob SDKs?

Reply:
This falls under the changes we discussed in person. Please review again the new API functions.